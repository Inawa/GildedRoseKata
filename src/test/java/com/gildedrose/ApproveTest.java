package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;
import org.approvaltests.Approvals;

public class ApproveTest {

    @Test
    public void test_approval() {
        Item[] items = new Item[] { new Item("foo", 0, 10) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        Approvals.verifyAll("", app.items);
    }

    @Test
    public void update_quality() {
        CombinationApprovals.verifyAllCombinations(
                (name, sellIn, quality) -> {
                    Item[] items = new Item[] { new Item(name, sellIn, quality) };
                    GildedRose app = new GildedRose(items);
                    app.updateQuality();
                    return app.items[0];
                },
                new String[] {"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"},
                new Integer[] { -1, 0, 20},
                new Integer[] { 0, 1, 49, 60 });

//        new String[] { "foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros" },
//            new Integer[] { -1, 0, 1, 5, 6, 10, 11 },
//            new Integer[] { 0, 1, 49, 50 });
    }
}
