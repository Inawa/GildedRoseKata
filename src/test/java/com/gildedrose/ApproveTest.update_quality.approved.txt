[foo, -1, 0] => foo, -2, 0 
[foo, -1, 1] => foo, -2, 0 
[foo, -1, 49] => foo, -2, 47 
[foo, -1, 60] => foo, -2, 58 
[foo, 0, 0] => foo, -1, 0 
[foo, 0, 1] => foo, -1, 0 
[foo, 0, 49] => foo, -1, 47 
[foo, 0, 60] => foo, -1, 58 
[foo, 20, 0] => foo, 19, 0 
[foo, 20, 1] => foo, 19, 0 
[foo, 20, 49] => foo, 19, 48 
[foo, 20, 60] => foo, 19, 59 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 60] => Aged Brie, -2, 60 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 60] => Aged Brie, -1, 60 
[Aged Brie, 20, 0] => Aged Brie, 19, 1 
[Aged Brie, 20, 1] => Aged Brie, 19, 2 
[Aged Brie, 20, 49] => Aged Brie, 19, 50 
[Aged Brie, 20, 60] => Aged Brie, 19, 60 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 60] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 60] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 20, 0] => Backstage passes to a TAFKAL80ETC concert, 19, 1 
[Backstage passes to a TAFKAL80ETC concert, 20, 1] => Backstage passes to a TAFKAL80ETC concert, 19, 2 
[Backstage passes to a TAFKAL80ETC concert, 20, 49] => Backstage passes to a TAFKAL80ETC concert, 19, 50 
[Backstage passes to a TAFKAL80ETC concert, 20, 60] => Backstage passes to a TAFKAL80ETC concert, 19, 60 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 60] => Sulfuras, Hand of Ragnaros, -1, 60 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 60] => Sulfuras, Hand of Ragnaros, 0, 60 
[Sulfuras, Hand of Ragnaros, 20, 0] => Sulfuras, Hand of Ragnaros, 20, 0 
[Sulfuras, Hand of Ragnaros, 20, 1] => Sulfuras, Hand of Ragnaros, 20, 1 
[Sulfuras, Hand of Ragnaros, 20, 49] => Sulfuras, Hand of Ragnaros, 20, 49 
[Sulfuras, Hand of Ragnaros, 20, 60] => Sulfuras, Hand of Ragnaros, 20, 60 
